### ISSUE TITLE
tensorflow backend and index-based-upsampling

### ISSUE THREAD

**Issue by opnumten** [imlab-uiip/keras-segnet#8]
Hi, thanks for sharing the code.
I have two questions:

1 tensorflow backend. I use tensorflow back to train the model, But the predicted output is poor.
2 how to use your index-based-upsampling code? If I use tensorflow backend, how do I modify it?

Thank you.

**Comment by eclique** [comment]
Hi, @opnumten.
In order to implement index based upsampling with tensorflow you would need to modify tf pooling function so that it would memorize max indices, similarly you would need custom upsampling function that would do upsampling with repsect to passed indices. Then you would need to implement custom layers performing this operations.
Code in index-based-upsampling does all that for old theano version.

**Comment by opnumten** [comment]
Thank you. For the 1st question, I use all the parameters you share except I use tensorflow backend, the result is : **_Test score: 0.661281108856_**
**_Test accuracy: 0.683624267578_**
I don't konw why.  I don't think backend should have such huge influence 
![result](https://cloud.githubusercontent.com/assets/26777414/24563629/20f40ee8-161d-11e7-8334-6ca12bca6e9b.png)

**Comment by PellyKoo** [comment]
@opnumten  We have the same problem, the result is so bad:
Test score: 0.645711183548
Test accuracy: 0.669636964798


![result](https://user-images.githubusercontent.com/34641707/39466320-e54395fa-4d5a-11e8-8185-459ec357ecc4.png)
